GNUmakefile

```diff
@@ -257,12 +251,18 @@ hclfmt:
 	@find . -path ./terraform -prune -o -name 'upstart.nomad' -prune -o \( -name '*.nomad' -o -name '*.hcl' \) -exec \
 sh -c 'hclfmt -w {} || echo in path {}' ';'
 
+.PHONY: sync
+sync:
+	@echo "--> Sync vendor directory and tidy up"
+	@go mod vendor
+	@go mod tidy
+
 .PHONY: dev
 dev: GOOS=$(shell go env GOOS)
 dev: GOARCH=$(shell go env GOARCH)
 dev: GOPATH=$(shell go env GOPATH)
 dev: DEV_TARGET=pkg/$(GOOS)_$(GOARCH)/nomad
-dev: vendorfmt changelogfmt hclfmt ## Build for the current development platform
+dev: sync changelogfmt hclfmt ## Build for the current development platform
```

Comment:
This seems to be a significant difference in behavior for something we probably run a lot in local development. When we were doing `vendorfmt` that's more the equivalent of `go mod tidy`, whereas now we have the side-effect of potentially making a bunch of changes to what's been vendored on every local build.

Reply:
Good catch @tgross ! Seems like we can maintain the previous behavior by breaking out this `sync` target into 2: `tidy` and `sync`. The `dev` target would only reference `tidy`, which would be close to `vendorfmt` (with the exception that `tidy` removes unused modules, but I think that's okay). `sync` would then just be a convenience target to run both `go mod tidy` and `go mod vendor`, which is handy when updating the `go.mod` with a new dependency/version.